### PR TITLE
doc: allow notes on Param API Reference pages

### PR DESCRIPTION
**Pull request by antoniotorresm**
The notes that Param pages will contain after #6733 are added manually, and because of it we need to add markers to differentiate between automated and manual content, equal to what we do for class pages.

### CODE REVIEW

File: makeapi.in [hunk 1]
```diff
@@ -331,9 +331,21 @@ def make_api_reference(validate_only=False):
 
     for param in ipa_classes:
         lines = [
+            "[//]: # (THE CONTENT BELOW IS GENERATED. DO NOT EDIT.)",
             ".. _%s:\n" % param.__name__,
             "# %s" % param.__name__,
+            "[//]: # (ADD YOUR NOTES BELOW. THESE WILL BE PICKED EVERY TIME THE DOCS ARE REGENERATED. //end)"  # noqa: E501
         ]
```
[the remark this answers is not in the document]
Comment (antoniotorresm) [reply makeapi.in:342]: Thanks, extracted the markers to variables.

**Comment by rcritten** [comment]
So to test this we  would apply this patch, then make some changes within the begin/end block and confirm that an API error is not thrown?

**Comment by antoniotorresm** [comment]
The only generated content in Param pages is the title, so it's the only thing inside begin/end. Change it and you should see errors, but adding notes below it should be allowed. Before the patch, any modification done to the file would fail the verification.

**Comment by rcritten** [comment]
I tested with changes:
- [x] before the comment block, correctly flagged
- [x] inside the comment block, a surprise to be flagged but it's fine
- [x] below the comment block and no error

Appears to to be working as designed.

**Comment by flo-renaud** [comment]
master:

* c553e3ceb280acf05064c6ff323d9666a406447f doc: allow notes on Param API Reference pages

**Comment by flo-renaud** [comment]
Hi @antoniotorresm 
the patch failed to apply cleanly on ipa-4-9, can you do a manual backport? Thanks

**Pull request closed without merging**